File: JavaScript/CounterProgram/counter.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let counter = 0;

function displayMenu() {
  console.log('\n========================');
  console.log(`Current Counter: ${counter}`);
  console.log('========================');
  console.log('1. Increase counter by 1');
  console.log('2. Decrease counter by 1');
  console.log('3. Exit');
  console.log('========================');
}

function handleChoice(choice) {
  switch (choice.trim()) {
    case '1':
      counter++;
      console.log(`Counter increased to: ${counter}`);
      promptUser();
      break;
    case '2':
      counter--;
      console.log(`Counter decreased to: ${counter}`);
      promptUser();
      break;
    case '3':
      console.log(`\nFinal counter value: ${counter}`);
      console.log('Thank you for using the Counter Program!');
      rl.close();
      break;
    default:
      console.log('Invalid choice. Please enter 1, 2, or 3.');
      promptUser();
      break;
  }
}

function promptUser() {
  displayMenu();
  rl.question('Enter your choice (1-3): ', handleChoice);
}

console.log('\n=== Interactive Counter Program ===');
console.log('Welcome! This program allows you to increase or decrease a counter.\n');
promptUser();

File: Rust/CounterProgram/counter.rs
use std::io::{self, Write};

fn main() {
    let mut counter = 0;
    
    println!("Welcome to the Rust Counter Program!");
    println!("Commands: [i]ncrement, [d]ecrement, [s]how, [e]xit");
    
    loop {
        print!("\nEnter command: ");
        io::stdout().flush().unwrap();
        
        let mut input = String::new();
        io::stdin().read_line(&mut input)
            .expect("Failed to read line");
        
        let command = input.trim().to_lowercase();
        
        match command.as_str() {
            "i" | "increment" => {
                counter += 1;
                println!("Counter incremented! Current value: {}", counter);
            }
            "d" | "decrement" => {
                counter -= 1;
                println!("Counter decremented! Current value: {}", counter);
            }
            "s" | "show" => {
                println!("Current counter value: {}", counter);
            }
            "e" | "exit" => {
                println!("Exiting... Final counter value: {}", counter);
                break;
            }
            _ => {
                println!("Invalid command! Use: [i]ncrement, [d]ecrement, [s]how, or [e]xit");
            }
        }
    }
}
